### PR TITLE
specification for pool list view

### PR DESCRIPTION
**Pull request by gnehapk**
@anivargi @brainfunked @nthomas-redhat @anmolbabu @kamleshverma1 Please review.

### CODE REVIEW

File: specs/pool_list_view.adoc [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+/api/1.0/GetPoolList
+
+Required Input from Anup - 
+
+* To add the response format for the API call
+* To add the polling API call and its response format
```
Comment (gnehapk) [inline specs/pool_list_view.adoc]: @anivargi Please provide the necessary inputs here.

File: specs/pool_list_view.adoc [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+/api/1.0/GetPoolList
+
+Required Input from Anup - 
+
+* To add the response format for the API call
+* To add the polling API call and its response format	
+
+==== Notifications/Monitoring impact:
+
+Required Input from Anmol: To add the response format for monitoring data
```
Comment (gnehapk) [inline specs/pool_list_view.adoc]: @anmolbabu Please provide the necessary inputs here.

**Comment by gnehapk** [comment]
Addresses #75.

**Comment by gnehapk** [comment]
Addresses #75.

File: specs/pool_list_view.adoc [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+* PGs
+* Quotas
+* Active alerts
+
+API support present for -
+
+* Pool Name
+* Status
+* Cluster Name
+* CPU Utilisation
```
Comment (cybernth) [inline specs/pool_list_view.adoc]: CPU utilization? or is it pool utilization?

File: specs/pool_list_view.adoc [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+Generate list component can be created and used for listing the pool. But because of its complexity and time constraint, we will pick this up later.
+
+=== Data model impact:
+
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+/api/1.0/GetPoolList
```
Comment (cybernth) [inline specs/pool_list_view.adoc]: This part lack information. state what is available currently(cluster wide list API available), what extra information reqd in the current API. Also specify the any extra APIs required(syatem wide listing?)
Which component implements sorting and filtering? 
Any APis required to get the utlization data?

File: specs/pool_list_view.adoc [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+== Testing:
+
+Test whether pools present in a ceph cluster are correctly being displayed or not.
+
+== Documentation impact:
+
+None
+
+== References:
+
```
Comment (cybernth) [inline specs/pool_list_view.adoc:145]: Refer the topic issue here - https://github.com/Tendrl/specifications/issues/75

File: specs/pool_list_view.adoc [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+* filter by pool name, cluster name and status.
+* sort by pool name.
+
+== Proposed change
+
+* Create a view for listing pool associated with a ceph cluster.
+* For utilisation, use donut chart component.
+* Implement client side filer and sorting functionality.
+* Filter call will trigger after 1 sec the user enters the filter data.
+* Send polling call after 10 minutes to get the updated data.
```
Comment (cybernth) [inline specs/pool_list_view.adoc]: 10 mins might not be acceptable

File: specs/pool_list_view.adoc [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+Tendrl should show list of pool associated with a ceph cluster along with their details.
+
+The details shown for each pool in UI is -
+
+* Pool Name
+* Status
+* Cluster Name
+* Pool Type
+* CPU Utilisation
```
Comment (cybernth) [inline specs/pool_list_view.adoc]: typo?

**Review by cybernth — CHANGES_REQUESTED** [review]

**Comment by gnehapk** [comment]
@cloudbehl @anivargi I have updated the PR. Please review it.

**Review by cloudbehl — COMMENTED** [review]
LGTM.
Need to decide upon the id's and classes for BDD(Behavior-driven development). I think we can create different spec which will have common id's and classes we will be using.

File: specs/pool_list_view.adoc [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+Tendrl should show list of pool associated with a ceph cluster along with their details.
+
+The details shown for each pool in UI is -
+
+* Pool Name
+* Status
+* Cluster Name
+* Pool Type
+* Utilisation
```
Comment (ltrilety) [inline specs/pool_list_view.adoc]: What utilization? Storage used?

**Comment by gnehapk** [comment]
@anivargi For the pool list view, below are the data that is not present in the json response of pool -

1) Status
2) Type
3) Utilisation
4) Replicas
5) OSDs
6) Quotas
7) Alerts

These are the data(with their mapped attribute in json) which are provided by API -

1) name - pools.poolname
2) Cluster name - maps.cluster_name
3) PGs - pools.pg_num

Please confirm, whether these mappings are correct or not.

**Comment by mkudlej** [comment]
Please add https://github.com/Tendrl/usmqe-tests/issues/17 as testing reference.

**Pull request merged**